src/loader/chunk_append.c

```diff
@@ -0,0 +1,56 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+
+#include <postgres.h>
+#include <fmgr.h>
+#include <miscadmin.h>
+#include <storage/lwlock.h>
+#include <storage/shmem.h>
+
+#include "loader/chunk_append.h"
+#include "export.h"
+
+#define CHUNK_APPEND_SHARED_STATE_NAME "ts_chunk_append_shared_state"
+#define CHUNK_APPEND_LWLOCK_TRANCHE_NAME "ts_chunk_append_lwlock_tranche"
+
+/*
+ * since shared memory can only be setup in a library loaded as
+ * shared_preload_libraries we have to setup this struct here
+ */
+typedef struct ChunkAppendSharedState
```

Comment:
Given that this wrapper struct seems to be necessary for lw locks, I wonder if it's worth making this general as in struct `TsLWLocks { LWLock *chunk_append_lock }` and then make the rendezvous variable point to the struct not the lock. This will make adding additional LWLocks much easier in the future.